minecraft-server/start-minecraft.sh

```diff
@@ -33,19 +33,35 @@ esac
 
 cd /data
 
-echo "Checking minecraft / forge type information."
+echo "Checking type information."
 case "$TYPE" in
-  VANILLA)
-    SERVER="minecraft_server.$VANILLA_VERSION.jar"
-
-    if [ ! -e $SERVER ]; then
-      echo "Downloading $SERVER ..."
-      wget -q https://s3.amazonaws.com/Minecraft.Download/versions/$VANILLA_VERSION/$SERVER
-    fi
-  ;;
+  *BUKKIT|*bukkit|SPIGOT|spigot)
+    TYPE=SPIGOT
+    case "$TYPE" in
+      *BUKKIT|*bukkit)
+        echo "Downloading latest CraftBukkit 1.8 server ..."
+        SERVER=craftbukkit_server.jar
+        ;;
+      *)
+        echo "Downloading latest Spigot 1.8 server ..."
+        SERVER=spigot_server.jar
+        ;;
+    esac
+    case $VANILLA_VERSION in
+      1.8*)
+        URL=/spigot18/$SERVER
+        ;;
+      *)
+        echo "That version of $SERVER is not available."
+        exit 1
+      ;;
+    esac
+    wget -q https://getspigot.org$URL
```

Comment:
The only pity is that there is no simple way to get Version $X or a list of available versions, though you can get Latest 1.8.  As it is, only v1.8, 1.8.3 and 1.8.8 are available via the Spigot Jenkins and are all compatible.